packages/fxa-settings/src/models/reliers/oauth-relier.ts

```diff
@@ -196,6 +193,35 @@ export class OAuthRelier extends BaseRelier implements OAuthRelierData, Relier {
     return this.service || this.clientId;
   }
 
+  getPermissions() {
+    // Ported from content server, search for _normalizeScopesAndPermissions
+    let permissions = scopeStrToArray(this.scope || '');
+
+    if (this.isTrusted()) {
+      // We have to normalize `profile` into is expanded sub-scopes
+      // in order to show the consent screen.
+      if (this.wantsConsent()) {
+        permissions = replaceItemInArray(
+          permissions,
+          Constants.OAUTH_TRUSTED_PROFILE_SCOPE,
+          Constants.OAUTH_TRUSTED_PROFILE_SCOPE_EXPANSION
+        );
```

Comment:
Not sure I'm tracking everything backwards far enough, but I might have a more serious concern about the `replaceItemInArray()` behavior of always appending the `replaceWith[]` regardless of whether `itemToReplace` was a hit or miss.

Q: Is it possible that this check will now _ALWAYS_ append the four `profile:*` permissions into the requested permissions regardless of whether the original `permissions` included `Constants.OAUTH_TRUSTED_PROFILE_SCOPE` permission?

https://github.com/mozilla/fxa/blob/89ce0d9e14ea825c2f4db285c15781af56d418ba/packages/fxa-settings/src/lib/constants.ts#L57-L63